internal/app/workers/accrual.go

```diff
@@ -0,0 +1,130 @@
+package workers
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/MrTomSawyer/loyalty-system/internal/app/interfaces"
+	"github.com/MrTomSawyer/loyalty-system/internal/app/logger"
+	"github.com/MrTomSawyer/loyalty-system/internal/app/models"
+	"io"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+func HandleOrders(orderRep interfaces.OrderRepository, maxWorkers int, tickerPeriod string, accrualHost string) {
+	var m sync.Mutex
+	workerPool := make(chan struct{}, maxWorkers)
+
+	go func() {
+		tPeriod, err := strconv.Atoi(tickerPeriod)
+		if err != nil {
+			logger.Log.Errorf("Invalid ticker period format: %v", err)
+			return
+		}
+
+		ticker := time.NewTicker(time.Duration(tPeriod) * time.Second)
+		for range ticker.C {
+			unhandledOrderIDs, err := orderRep.GetUnhandledOrders()
+			if err != nil {
+				logger.Log.Errorf("Failed to get all ")
+			}
+			for _, orderID := range unhandledOrderIDs {
+				logger.Log.Infof("Procceeding order №%s", orderID)
+				workerPool <- struct{}{}
+
+				go func(orderID string) {
+					m.Lock()
+					defer m.Unlock()
+					defer func() {
+						<-workerPool
+					}()
+
+					userID, err := orderRep.GetOrderAndUserIDs(orderID)
+					if err != nil {
+						logger.Log.Errorf("no order with id=%s found: %v", orderID, err)
+						return
+					}
+
+					order := models.Order{}
+					order.OrderID = orderID
+					getAccrual(&order, accrualHost)
+					if order.Status == "" {
+						logger.Log.Infof("No accrual received for order №%s", orderID)
+						return
+					}
+
+					err = orderRep.UpdateOrderAccrual(order, orderID, userID)
+					if err != nil {
+						logger.Log.Errorf("failed to update order: %s", orderID)
+						return
+					}
+				}(orderID)
```

Comment:
```suggestion
				go processOrder(orderID)
```

тут бы легче читалось если разделить логику - текущая функция отвечала бы за получение заказов, и периодический запуск обработки заказа. А вторая функция только обрабатывала один конкретный заказ

Каждая функция будет выполнять небольшую понятную работу